Add certificate request

cwjenkins opened the pull request:
Add MarshalDER to certificateRequest
Add LoadCertificateRequestFromDER

openface commented on cert.go:
```diff
@@ -273,6 +322,20 @@ func (c *Certificate) Sign(privKey PrivateKey, digest EVP_MD) error {
 	return c.insecureSign(privKey, digest)
 }
 
+// Sign a certificate using a private key and a digest name.
```
Shouldn't this comment read "Sign a certificate **request** using a private key and a digest name."

cwjenkins replied:
That it should. Good catch, a lot of this is copied and pasted.

cwjenkins committed:
Add certificate request
Add MarshalDER to certificateRequest
Add LoadCertificateRequestFromDER
Add Free given intermittent failures with Finalizer

cwjenkins committed:
Change module name
